internal/provider/query_specification_data_source.go

```diff
@@ -89,6 +89,11 @@ func (d *querySpecDataSource) Schema(_ context.Context, _ datasource.SchemaReque
 				Optional:   true,
 				Validators: []validator.Int64{int64validator.AtLeast(0)},
 			},
+			"compare_time_offset_seconds": schema.Int64Attribute{
+				Description: "The time offset for comparison queries, in seconds. " +
+					"Used to compare current time range data with data from a previous time period.",
+				Optional: true,
```

Comment:
This should have a validator checking the possible values:

```suggestion
				Optional: true,
				Validators: []validator.Int64{int64validator.OneOf(
					1800, 3600, 7200, 14400, 28800, 86400, 604800, 2419200, 15724800,
				)},
```

Reply:
@jharley I didnt include a validator here because a possible value is also whatever the queries time range is (say the user set the time range to `10`, then `10` is also a possible value) so I figured we'd let the API handle the validation logic - is that the right way to go about it 🤔

Reply:
👀 ty for that insight, let me bring that validator logic here
(do we have a way to tell what requests come from terraform? i'd love to see our usage)